src/devices/cpu/tms32031/dis32031.cpp

```diff
@@ -122,62 +122,65 @@ static const char *const condition[32] =
-
-		case 16:    sprintf(dst, "*+AR%d(IR1)", ar);    break;
-		case 17:    sprintf(dst, "*-AR%d(IR1)", ar);    break;
-		case 18:    sprintf(dst, "*++AR%d(IR1)", ar);   break;
-		case 19:    sprintf(dst, "*--AR%d(IR1)", ar);   break;
-		case 20:    sprintf(dst, "*AR%d++(IR1)", ar);   break;
-		case 21:    sprintf(dst, "*AR%d--(IR1)", ar);   break;
-		case 22:    sprintf(dst, "*AR%d++(IR1)%%", ar); break;
-		case 23:    sprintf(dst, "*AR%d--(IR1)%%", ar); break;
-
-		case 24:    sprintf(dst, "*AR%d", ar);          break;
-		case 25:    sprintf(dst, "*AR%d++(IR0)B", ar);  break;
+		case 0:     util::stream_format(stream, "*+AR%d%s", ar, dispstr);  break;
+		case 1:     util::stream_format(stream, "*-AR%d%s", ar, dispstr);  break;
+		case 2:     util::stream_format(stream, "*++AR%d%s", ar, dispstr); break;
+		case 3:     util::stream_format(stream, "*--AR%d%s", ar, dispstr); break;
+		case 4:     util::stream_format(stream, "*AR%d++%s", ar, dispstr); break;
+		case 5:     util::stream_format(stream, "*AR%d--%s", ar, dispstr); break;
+		case 6:     util::stream_format(stream, "*AR%d++%s%%", ar, dispstr);   break;
+		case 7:     util::stream_format(stream, "*AR%d--%s%%", ar, dispstr);   break;
+
+		case 8:     util::stream_format(stream, "*+AR%d(IR0)", ar);    break;
+		case 9:     util::stream_format(stream, "*-AR%d(IR0)", ar);    break;
+		case 10:    util::stream_format(stream, "*++AR%d(IR0)", ar);   break;
+		case 11:    util::stream_format(stream, "*--AR%d(IR0)", ar);   break;
+		case 12:    util::stream_format(stream, "*AR%d++(IR0)", ar);   break;
+		case 13:    util::stream_format(stream, "*AR%d--(IR0)", ar);   break;
+		case 14:    util::stream_format(stream, "*AR%d++(IR0)%%", ar); break;
+		case 15:    util::stream_format(stream, "*AR%d--(IR0)%%", ar); break;
+
+		case 16:    util::stream_format(stream, "*+AR%d(IR1)", ar);    break;
+		case 17:    util::stream_format(stream, "*-AR%d(IR1)", ar);    break;
+		case 18:    util::stream_format(stream, "*++AR%d(IR1)", ar);   break;
+		case 19:    util::stream_format(stream, "*--AR%d(IR1)", ar);   break;
+		case 20:    util::stream_format(stream, "*AR%d++(IR1)", ar);   break;
+		case 21:    util::stream_format(stream, "*AR%d--(IR1)", ar);   break;
+		case 22:    util::stream_format(stream, "*AR%d++(IR1)%%", ar); break;
+		case 23:    util::stream_format(stream, "*AR%d--(IR1)%%", ar); break;
+
+		case 24:    util::stream_format(stream, "*AR%d", ar);          break;
+		case 25:    util::stream_format(stream, "*AR%d++(IR0)B", ar);  break;
 		case 28:
 		case 29:
 		case 30:
-		case 31:    strcpy(dst, regname[ma & 31]);      break;
-		default:    sprintf(dst, "(unknown mode)");     break;
+		case 31:    stream << regname[ma & 31];      break;
+		default:    util::stream_format(stream, "(unknown mode)");     break;
 	}
 }
 
-static void append_immediate(uint16_t data, int is_float, int is_unsigned, char *buffer)
+static void append_indirect(uint8_t ma, int8_t disp, std::string &string)
```

Comment:
Calling this `append_indirect` when it actually overwrites the destination string rather than appending to it is misleading.